recipe/meta.yaml

```diff
@@ -1,46 +1,54 @@
-{% set version = "1.42.0" %}
+{% set version = "1.62.2" %}
 {% set name = "grpcio-tools" %}
 
 package:
-  name: "{{ name|lower }}"
-  version: "{{ version }}"
+  name: {{ name|lower }}
+  version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: d0a0daa82eb2c2fb8e12b82a458d1b7c5516fe1135551da92b1a02e2cba93422
-  patches:
-    - 0001-Monkey-patch-distutils.ccompiler.spawn-to-elide-std-.patch
+  sha256: 5fd5e1582b678e6b941ee5f5809340be5e0724691df5299aae8226640f94e18f
+  # more work required to unvendor win - to be done in later update.
+  patches:                          # [not win]
+   - unvendor_abseil_protobuf.patch # [not win]
 
 build:
-  number: 2
+  skip: true  # [py<37]
+  number: 0
   missing_dso_whitelist:  # [s390x]
     - $RPATH/ld64.so.1    # [s390x]
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - patch       # [not win]
-    - m2-patch    # [win]
- 
+    - patch # [unix]
+    - m2-patch # [win]
   host:
-    - grpcio =={{ version }}
     - pip
     - setuptools
     - wheel
-    - protobuf 3.20.3
     - python
-    - cython 0.29.32
+    - cython
+    - libabseil 20240116.2
+    - libprotobuf 4.25.3
+    - protobuf 4.25.3
   run:
-    - grpcio =={{ version }}
-    - protobuf >=3.6.0
+    - grpcio >={{ version }}
+    - protobuf >=4.21.6,<5.0a0
+    - setuptools
     - python
+    # through run_exports
+    - libabseil
```

Comment:
libabseil is not needed. The only abseil type used directly is `absl::string_view` (https://github.com/grpc/grpc/blob/master/tools/distrib/python/grpcio_tools/grpc_tools/main.cc#L114-L121) which is an alias to `std::string_view` in our case. This means that at the library will use the std symbols and nor the abseil symbols:
```
(test) jcmorin-mac-anac:work jcmorin$ nm ../_h_env/lib/python3.12/site-packages/grpc_tools/_protoc_compiler.cpython-312-darwin.so | c++filt | grep -e RecordError -e RecordWarning
000000000002492c t grpc_tools::internal::ErrorCollectorImpl::RecordError(std::__1::basic_string_view<char, std::__1::char_traits<char>>, int, int, std::__1::basic_string_view<char, std::__1::char_traits<char>>)
0000000000024b34 t grpc_tools::internal::ErrorCollectorImpl::RecordWarning(std::__1::basic_string_view<char, std::__1::char_traits<char>>, int, int, std::__1::basic_string_view<char, std::__1::char_traits<char>>)
```

So we can use `ignore_run_exports` on `libabseil` to ignore the DSO warning.

Reply:
As `grpcio-tools` relies on `-std=c++14` https://github.com/grpc/grpc/blob/v1.62.2/tools/distrib/python/grpcio_tools/setup.py#L93 then `libabseil` is not needed but for `c++17` and `protobuf >=22` I guess it will be required, see https://github.com/opencv/opencv/issues/24369